pyanaconda/input_checking.py

```diff
@@ -0,0 +1,795 @@
+    def status_text(self):
+        """A short overall status message describing the password.
+
+        Generally something like "Good.", "Too short.", "Empty.", etc.
+
+        :rtype: short status message
+        :rtype: str
+        """
+        return self._status_text
+
+    @status_text.setter
+    def status_text(self, new_status_text):
+        self._status_text = new_status_text
+        self.status_text_changed.emit(new_status_text)
+
+    @property
+    def password_quality(self):
+        """More fine grained integer indicator describing password strength.
+
+        This basically exports the quality score assigned by libpwquality to the password,
+        which goes from 0 (unacceptable password) to 100 (strong password).
+
+        Note of caution though about using the password quality value - it is intended
+        mainly for on-line password strength hints, not for long-term stability,
+        even just because password dictionary updates and other peculiarities of password
+        strength judging.
+
+        :returns: password quality value as reported by libpwquality
+        :rtype: int
+        """
+        return self._password_quality
+
+    @password_quality.setter
+    def password_quality(self, value):
+        self._password_quality = value
+        self.password_quality_changed.emit(value)
+
+    @property
+    def length_ok(self):
+        """Reports if the password is long enough.
+
+        :returns: if the password is long enough
+        :rtype: bool
+        """
+        return self._length_ok
+
+    @length_ok.setter
+    def length_ok(self, value):
+        self._length_ok = value
+        self.length_ok_changed.emit(value)
+
+class InputCheck(object):
+    """Input checking base class."""
+
+    def __init__(self):
+        self._result = CheckResult()
+        self._skip = False
+
+    @property
+    def result(self):
+        return self._result
+
+    @property
+    def skip(self):
+        """A flag hinting if this check should be skipped."""
+        return self._skip
+
+    @skip.setter
+    def skip(self, value):
+        # Checks flagged as skipped are
+        # considered successful.
+        # Otherwise old state will linger even if the
+        # check is skipped during checking runs.
+        if value:
+            self.result.error_message = ""
+            self.result.success = True
+        self._skip = value
+
+    def run(self, check_request):
+        """Run the check.
+
+        :param check_request: arbitrary input data to be processed
+
+        Subclasses need to always implement this.
+        """
+        raise NotImplementedError
+
+class RegexpCheck(InputCheck):
+    """A regex based input check."""
+
+    def __init__(self, regexp, error_message):
+        """
+        :param regexp: a regular expression object
+        :param error_message: error message to return if the regexp doesn't match
+        """
+        super().__init__()
+        self._result = CheckResult()
```

Comment:
Shouldn't this already have been set in the parent's init?

Reply:
Indeed, this is redundant unless you want to use a different result class instance. :)